src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -598,36 +623,45 @@ protected Expr createTransportSettingsInitExpr(
     createBody.add(ExprStatement.withExpr(idComponentsAssignExpr));
 
     // Generate return statement
-    // This will be replaced and edited based on annotations
-    TypeNode getRegionOperationRequestType =
-        TypeNode.withReference(
-            VaporReference.builder()
-                .setName("GetRegionOperationRequest")
-                .setPakkage("com.google.cloud.compute.v1")
-                .setIsStaticImport(false)
-                .build());
-    MethodInvocationExpr newBuilderExpr =
-        MethodInvocationExpr.builder()
-            .setStaticReferenceType(getRegionOperationRequestType)
-            .setMethodName("newBuilder")
-            .build();
-    newBuilderExpr =
-        methodMaker
-            .apply("setOperation", Collections.singletonList(getExpr(idComponentsVarExpr, "0")))
-            .apply(newBuilderExpr);
-    newBuilderExpr =
-        methodMaker
-            .apply("setProject", Collections.singletonList(getExpr(idComponentsVarExpr, "1")))
-            .apply(newBuilderExpr);
-    newBuilderExpr =
-        methodMaker
-            .apply("setRegion", Collections.singletonList(getExpr(idComponentsVarExpr, "2")))
-            .apply(newBuilderExpr);
+    TypeNode getOperationRequestType = TypeNode.withReference(protoMethod.inputType().reference());
+    MethodInvocationExpr newBuilderExpr = staticBuilder(getOperationRequestType);
+    Map<String, String> responseFields = inputOperationMessage.operationResponseFields();
+    // TODO: revise variable name
+    List<String> responseFieldAnnotationNames = new ArrayList<String>(responseFields.keySet());
+    Collections.sort(responseFieldAnnotationNames);
+    ImmutableSet<String> fieldNames = inputOperationMessage.fieldMap().keySet();
+    ArrayList<String> nonResponseAnnotationFields = new ArrayList<String>();
+    for (String fieldName : fieldNames) {
+      if (!responseFields.containsValue(fieldName)) {
```

Comment:
This is not correct usage of maps. `map.containsValue()` will give you the linear time (do not confuse with containsKey(), which will give amortized O(1)).

This is not critical for this application, since there will never be many elements in the map, but basically if you need both keys and values to be indexed, you may consider using two-way index (direct and reverse, assuming that values are also unique).

Reply:
![image](https://user-images.githubusercontent.com/26678939/131572543-f11d4b14-7011-4ea1-96d9-de35122bce48.png) I created a collection used regular contains. I tried to find the runtime for this and couldn't find it, but I believe it is O(1)

Reply:
`Collection` is just an interface it can have many implementations. Here you did the same thing but in a more convoluded synax: `responseFieldName = map.values()` - which is still a non-indexed "array" of values. In other words, the proposed solution is exactly the same as it was, but a more verbose. 

In IntelliJ you can go to the SDK library classes implementations the same way as you go to your code implementation (ctrl + lef_mouse_click in most basic case). Reading the code is usually the best way to asses how it works and which time properties it has.

Basically only key operations are O(1) in maps. 

Side note: Please consider posting code as text instead of posting as a picture (it makes it mucheasier to work with

There is a way to post it higlighted:
```java
public static void main(String[] args) {}
```